src/parser/warrior/fury/modules/talents/FrothingBerserker.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+
+import Analyzer from 'parser/core/Analyzer';
+
+import SPELLS from 'common/SPELLS/index';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+import StatisticBox from 'interface/others/StatisticBox';
+
+/**
+ Frothing Berserker users should maximize casts with 100 rage to proc Frothing Berserker the post
+ Carnage users should cast as soon as possible, i.e. at 70 rage
+ Massacre users (not from Soul of the Battlelord) should cast as soon as possible, i.e. at 85 rage
+ Whenever Battle Cry is active, user should cast Rampage whenever possible - regardless of talents
+ */
+
+class FrothingBerserker extends Analyzer {
+
+  last_timestamp = null;
+  casts_counter = 0;
+  premature_counter = 0;
+  reckless_abandon = null;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.FROTHING_BERSERKER_TALENT.id);
+    this.reckless_abandon = this.selectedCombatant.hasTalent(SPELLS.RECKLESS_ABANDON_TALENT.id);
+  }
+
+  on_byPlayer_cast(event) {
```

Comment:
While not required (as in you don't have to resolve this, just a suggestion), the logic I see in this event makes me believe it would benefit greatly from the new event listeners. Take a look at #2705 (L34-38) for an example of how you can split up event listeners for specific spells and help lower the number of nested conditionals.

Reply:
I'll have to rewrite this module anyway, so i've just removed it for now